gef.py

```diff
@@ -3716,41 +3717,65 @@ def parsed_arglist(arglist):
 
     def synchronize(self):
         """Submit all active breakpoint addresses to IDA/BN"""
-        breakpoints = gdb.breakpoints() or []
-        old_bps = []
-
-        for x in breakpoints:
-            if x.enabled and not x.temporary:
-                val = gdb.parse_and_eval(x.location)
-                addr = str(val).strip().split()[0]
-                addr = long(addr, 16)
-                old_bps.append(addr)
 
         pc = current_arch.pc
+
+        vmmap = get_process_maps()
+        base_address = min([x.page_start for x in vmmap if x.path == get_filepath()])
+        end_address = max([x.page_end for x in vmmap if x.path == get_filepath()])
+        if pc > end_address or pc < base_address:
+            # do not sync in library
+            return
+
+        gdb_bps = []
+        for l in gdb.execute("info break", to_string=True).splitlines():
```

Comment:
If I understood correctly, it is not a problem or a bug, `gdb.Breakpoint.location` is a string, not a `gdb.Value`. So a more Python approach I believe would be something like:
```python
for bp in gdb.breakpoints():
 if bp.location[0]=='*' and is_hex(bp.location[0]):  # if it's an address i.e. location starts with '*'
  addr = long(bp.location[0], 16)
 else: # it is a symbol
  addr = long( gdb.parse_and_eval(bp.location).address )
 [...]
```

Would that be closer to what you intend to do?